csrc/quantization/compressed_tensors/int8_quant_kernels.cu

```diff
@@ -0,0 +1,56 @@
+#include <ATen/cuda/CUDAContext.h>
+#include <torch/extension.h>
+#include <assert.h>
+
+#include "../../dispatch_utils.h"
+
+static inline __device__ int8_t float_to_int8_rn(float x) {
+#ifdef USE_ROCM
+  float dst;
+  // Round to nearest even
+  asm volatile("v_rndne_f32 %0, %1;" : "=v"(dst) : "v"(x));
+  // Saturate
+  dst = dst < -128.0f ? -128.0f : dst;
+  dst = dst > 127.0f ? 127.0f : dst;
+  return static_cast<int8_t>(dst);
+#else
+  uint32_t dst;
+  asm volatile("cvt.rni.sat.s8.f32 %0, %1;" : "=r"(dst) : "f"(x));
+  return reinterpret_cast<const int8_t&>(dst);
+#endif
+}
+
+namespace vllm {
+
+template <typename scalar_t, typename scale_type>
+__global__ void static_scaled_int8_quant_kernel(
+    const scalar_t* __restrict__ input, int8_t* __restrict__ out,
+    scale_type scale, const int hidden_size) {
+  const int tid = threadIdx.x;
+  const int token_idx = blockIdx.x;
+
+  for (int i = tid; i < hidden_size; i += blockDim.x) {
+    out[token_idx * hidden_size + i] =
+        float_to_int8_rn(((float)input[token_idx * hidden_size + i]) / scale);
+  }
+}
+}  // namespace vllm
+
+void static_scaled_int8_quant(torch::Tensor& out,    // [..., hidden_size]
+                              torch::Tensor& input,  // [..., hidden_size]
+                              float scale) {
+  assert(input.is_contiguous());
```

Comment:
both of these asserts should be TORCH_CHECK so the interpreter doesn't crash if this gets triggered

Reply:
Done :+1: